core/app/Main.hs

```diff
@@ -17,13 +13,12 @@ import Data.Proxy
 address :: String
 address = "0.0.0.0"
 
+port :: Int 
+port = 3000
+
 main = do
   filepath : _ <- getArgs
   state <- initializeState filepath
   nb <- loadNotebookFromFile filepath
-  let server :: Server API = return filepath
-  run 3000 (websocketsOr WS.defaultConnectionOptions (application filepath nb state) (serve api server))
+  WS.runServer address port (application filepath nb state)
```

Comment:
That could be nice to know exactly why Servant was failing, but anyway I do like this line more, IMO it's clearer